docs/advanced/document-store/subindex.md

```diff
@@ -0,0 +1,159 @@
+(subindex)=
+# Subindices
+
+Subindices allow you to efficiently search through multimodal or nested Documents.
+
```

Comment:
should we add a link here to the example in text/index.md ? So that people who end up on this page can understand why the feature is needed in some case

Reply:
or the multi-modal one, yep, I agree